src/ucp/rndv/proto_rndv.c

```diff
@@ -76,6 +77,11 @@ ucp_proto_rndv_ctrl_get_md_map(const ucp_proto_rndv_ctrl_init_params_t *params,
             continue;
         }
 
+        if (!ucp_memh_sys_dev_reachable(
+                  mem_sys_dev, context->tl_mds[md_index].sys_dev_map)) {
```

Comment:
shouldn't we check the sys_dev of the lane, instead of the md?

Reply:
But in of rndv we do know which device is going to be used, since it's used with a particular ep

Reply:
pushed in place (PR is small)